core/src/test/java/uk/gov/gchq/koryphe/tuple/function/TupleAdaptedFunctionCompositeTest.java

```diff
@@ -0,0 +1,142 @@
+                                "\"by\": 10" +
+                            "}," +
+                            "\"projection\": [ \"output\" ]" +
+                        "}" +
+                    "]" +
+                "}";
+        // When
+        String serialised = JsonSerialiser.serialise(instance);
+        TupleAdaptedFunctionComposite deserialised = JsonSerialiser.deserialise(json, TupleAdaptedFunctionComposite.class);
+
+        // Then
+        JsonSerialiser.assertEquals(json, serialised);
+        assertEquals(instance, deserialised);
+    }
+
+    @Override
+    protected TupleAdaptedFunctionComposite<String> getInstance() {
+        return new TupleAdaptedFunctionComposite.Builder<String>()
+                .select(new String[] { "input" })
+                .execute(new ToLong())
+                .project(new String[] { "midway"})
+                .select(new String[] { "midway"})
+                .execute(new MultiplyLongBy(10))
+                .project(new String[] { "output"})
+                .build();
+    }
+
+    @Override
+    protected Iterable<TupleAdaptedFunctionComposite> getDifferentInstances() {
+        return Arrays.asList(
+                new TupleAdaptedFunctionComposite.Builder<String>()
+                        .select(new String[] { "DifferentInput" })
+                        .execute(new ToLong())
+                        .project(new String[] { "output"})
+                        .build(),
+                new TupleAdaptedFunctionComposite.Builder<String>()
+                        .select(new String[] { "input" })
+                        .execute(new ToString())
+                        .project(new String[] { "output"})
+                        .build(),
+                new TupleAdaptedFunctionComposite.Builder<String>()
+                        .select(new String[] { "input" })
+                        .execute(new ToLong())
+                        .project(new String[] { "differentOutput"})
+                        .build(),
+                new TupleAdaptedFunctionComposite.Builder<Integer>()
+                        .select(new Integer[] { 1 })
+                        .execute(new ToLong())
+                        .project(new Integer[] { 2 })
+                        .build(),
+                new TupleAdaptedFunctionComposite.Builder<String>()
+                        .build(),
+                new TupleAdaptedFunctionComposite.Builder<String>()
+                        .select(new String[] { "input" })
+                        .execute(new ToLong())
+                        .project(new String[] { "output"})
+                        .build()
+
+        );
+    }
+
+    @Test
+    public void shouldThrowExceptionIfInputIsWrongType() {
+        // Given
+        TupleAdaptedFunctionComposite<String> instance = getInstance();
+        MapTuple<String> inputTuple = new MapTuple<>();
+        inputTuple.put("input", "aString");
+
+        // When / Then
+        try {
```

Comment:
consider using assertThrows()

Reply:
Does assertThrows allow for error message checking?